src/ufuncs.c

```diff
@@ -90,6 +90,36 @@ static char GEOSisClosedAllTypes_r(void *context, void *geom) {
 }
 static void *is_closed_data[1] = {GEOSisClosedAllTypes_r};
 static void *is_valid_data[1] = {GEOSisValid_r};
+
+#if GEOS_SINCE_3_7_0
+static char GEOSGeom_isCCW_r(void *context, void *geom) {
+    const GEOSCoordSequence *coord_seq;
+    char is_ccw = 2;  // return value of 2 means GEOSException
+    int i;
+
+    // Return False for non-linear geometries
+    i = GEOSGeomTypeId_r(context, geom);
+    if (i == -1) { return 2; }
+    if ((i != GEOS_LINEARRING) & (i != GEOS_LINESTRING)) {
+        return 0;
+    }
+
+    // Return False for lines with fewer than 4 points
+    i = GEOSGeomGetNumPoints_r(context, geom);
+    if (i == -1) { return 2; }
+    if (i < 4) {
+        return 0;
+    }
```

Comment:
`GEOSCoordSeq_isCCW_r` doesn't handle this itself? Does it fail on shorter coordinate sequences?

Reply:
Actually, on second thought, do we need to check this? As `GEOSCoordSeq_isCCW_r` doesn't require the line to be a closed ring, so then the requirement of at least 4 coordinates might also not be needed?

Reply:
I put this here because it will else give the following exception:

```
In [2]: pygeos.is_ccw(pygeos.Geometry("LINESTRING (0 0, 0 1, 1 0)"))                                                                                                         
GEOSException: IllegalArgumentException: Ring has fewer than 3 points, so orientation cannot be determined
```

Note the number 3 in the exception message.

Reply:
Ah, OK, then indeed good to check for this. Maybe add a short comment about this that GEOS will error for <= 3 points and we want to return False?